EpiAware/src/latent-processes.jl

```diff
@@ -1,24 +1,41 @@
 abstract type AbstractLatentProcess end
+abstract type AbstractLatentProcessArg end
 
 struct RandomWalkLatentProcess{D <: Sampleable, S <: Sampleable} <: AbstractLatentProcess
     init_prior::D
     var_prior::S
 end
 
+struct RandomWalkLatentProcessArg <: AbstractLatentProcessArg end
+
 function default_rw_priors()
     return (:var_RW_prior => truncated(Normal(0.0, 0.05), 0.0, Inf),
         :init_rw_value_prior => Normal()) |> Dict
 end
 
-function generate_latent_process(latent_process::AbstractLatentProcess, n; kwargs...)
-    @info "No concrete implementation for generate_latent_process is defined."
+function latent_process(lp::AbstractLatentProcess, n;
+        kwargs...)
+    return latent_process(
+        AbstractLatentProcessArg(), n; var_prior = lp.var_prior, init_prior = lp.init_prior)
+end
+
+function latent_process(lp::AbstractLatentProcess, n; kwargs...)
+    @info "No concrete implementation for latent_process is defined."
     return nothing
 end
 
-@model function generate_latent_process(latent_process::RandomWalkLatentProcess, n)
+function latent_process(lp::RandomWalkLatentProcess, n)
+    return latent_process(
+        RandomWalkLatentProcessArg(), n; var_prior = lp.var_prior,
+        init_prior = lp.init_prior
+    )
+end
+
+@model function latent_process(lp::RandomWalkLatentProcessArg, n;
```

Comment:
It feels like the point of the above is to lower down to this function; but you can get the same effect by specifying the fields of `RandomWalkLatentProcess` (i.e. not using a `NamedTuple` or `Dict` of priors of potentially any name) and dispatching on that.

I'd avoid that by having `RandomWalkLatentProcess` have determined typed fields and saving a bunch of code repetition.